src/Bicep.Cli/Commands/ValidateParamsCommand.cs

```diff
@@ -0,0 +1,97 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Bicep.Cli.Arguments;
+using Bicep.Cli.Logging;
+using Bicep.Cli.Services;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Emit;
+using Bicep.Core.Features;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.TypeSystem;
+using Bicep.Core.Workspaces;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Bicep.Cli.Commands
+{
+    public class ValidateParamsCommand : ICommand
+    {
+        private readonly ILogger logger;
+        private readonly IDiagnosticLogger diagnosticLogger;
+        private readonly CompilationService compilationService;
+
+        public ValidateParamsCommand(
+            ILogger logger,
+            IDiagnosticLogger diagnosticLogger,
+            CompilationService compilationService)
+        {
+            this.logger = logger;
+            this.diagnosticLogger = diagnosticLogger;
+            this.compilationService = compilationService;
+        }
+
+        public async Task<int> RunAsync(ValidateParamsArguments args)
+        {
+            var bicepFilePath = PathHelper.ResolvePath(args.BicepFile);
+
+            if (!IsBicepFile(bicepFilePath))
+            {
+                logger.LogError(CliResources.UnrecognizedBicepFileExtensionMessage, bicepFilePath);
+                return 1;
+            }
+
+            var bicepCompilation = await compilationService.CompileAsync(bicepFilePath, false);
+
+            var parameterDeclarations = bicepCompilation.GetEntrypointSemanticModel().Parameters;
+
+            var parametersJson = JObject.Parse(args.Parameters);
+
+            var validationDiagnostics = ToListDiagnosticWriter.Create();
+
+            foreach(var parameter in parametersJson.Properties())
+            {   
+                if(parameterDeclarations.TryGetValue(parameter.Name, out var parameterMetadata))
+                {
+                    var declaredType = parameterMetadata.TypeReference.Type;
+                    var assignedType = SystemNamespaceType.ConvertJsonToBicepType(parameter.Value);
+
+                    if(!TypeValidator.AreTypesAssignable(assignedType, declaredType))
+                    {
+                        var diagnostic = DiagnosticBuilder
+                                        .ForDocumentStart()
+                                        .InvalidParameterValueAssignmentType(parameter.Name, declaredType);
+                        validationDiagnostics.Write(diagnostic);
+                    }
+                }  
+                else
+                {
+                    var diagnostic = DiagnosticBuilder
+                                    .ForDocumentStart()
+                                    .ParameterNotPresentInTemplate(parameter.Name, bicepFilePath);
+                        validationDiagnostics.Write(diagnostic);
+                } 
+            }
```

Comment:
I don't think we will run into that scenario because this command would be invoked by az cli/pwsh after it has requested the user to provide values for all required parameters